### DISCUSSION TITLE
Can suspense be used safely without react-dom@experimental?

### DISCUSSION THREAD

**Discussion by nggonzalez** [bvaughn/suspense#39] [Q&A]
Hi, does the `suspense` package actually have a dependency on `react-dom@experimental` or just `react@experimental`? I looked around the codebase and didn't see any `react-dom` imports in non test files, so I think it is safe, but wanted to double check in case I'm missing something.

I created a quick POC in an Expo app using SDK 49 and the RN `0.72.3`, and things seem to be working fine with only the `react@experimental` package installed

https://github.com/bvaughn/suspense/assets/5404649/16f1f4e2-ac70-4b72-a3c1-8d629eff5d7e

Very excited to use this package for my current RN project if only `react@experimental` is required 🤞

**Comment by bvaughn** [comment]
I believe you're correct that the only actual dependency is on the `react` package.

**Comment by nggonzalez** [comment]
Unfortunately running into some issues once I add `useCacheMutation`. Getting the following error:
```
TypeError: dispatcher.useCacheRefresh is not a function (it is undefined)
```

Is the dispatcher part of the platform renderer?

**Reply by bvaughn** [reply]
Ah, rats. Yes, the `ReactCurrentDispatcher.current` implementation is updated/injected by the reconciler (which gets bundled into the renderer). I guess that means you can't really use this with React Native, since there's no `react-native@experimental` package to depend on. (I'm not super familiar with the release process for RN, or how these experimental APIs, factor in.)

The RFC for `useCacheRefresh` makes no mention of React Native either so I'm not sure what the plans are.

**Reply by nggonzalez** [reply]
Ah so close! My guess is it'll land at some point since `getCacheForType` is already supported, but RN is typically behind by at least a few months. 

Thanks for the explanation. I'll keep an eye out and do some more research and update with whatever I find
